GccBoardProject1/src/peopleCounting.c

```
@@ -8,139 +8,164 @@
+	if (current_max < frame_conv[row+1][col-1] || current_max < frame_conv[row-1][col+1]) {
 		return false;
 	}
 	return true;
 }
 
-uint16_t *** COUNT_BUFF addFrame(uint16_t ** pix_buff, uint16_t*** COUNT_BUFF, uint16_t frameCount) {
+uint16_t* addFrame(uint16_t PIXEL_BUFFER[], uint16_t COUNT_BUFF[], uint16_t frameCount, struct keepCount *count) {
 	if (frameCount < 7) {
-		COUNT_BUFF[frameCount-1] = pix_buff;
+		for (int i = 0; i < NUM_PIXELS; i++) {
+			COUNT_BUFF[(frameCount-1)*NUM_PIXELS+i] = PIXEL_BUFFER[i];
+		}
 	}
 	else {
-		for (int i = 0; i < 6; i++){
-			COUNT_BUFF[i] = COUNT_BUFF[i+1];
+		for (int i = 0; i < 6*NUM_PIXELS; i++){
+			COUNT_BUFF[i] = COUNT_BUFF[NUM_PIXELS+i];
+		}
+		for (int i = 0; i < NUM_PIXELS; i++) {
+			COUNT_BUFF[7*NUM_PIXELS+i] = PIXEL_BUFFER[i];
 		}
-		COUNT_BUFF[7] = pix_buff;
 	}
+	count->counted = false;
 	return COUNT_BUFF;
 }
 
-uint16_t getMaxId(uint16_t ** array, uint16_t col) {
-	uint16_t maxNum = array[0];
+uint16_t getMaxId(uint16_t COUNT_BUFF[], uint16_t ind, uint16_t col) {
+	convertToFrame(COUNT_BUFF, ind);
+	
+	uint16_t maxNum = frame_conv[0][0];
 	uint16_t maxRowId = 0;
 	
 	for (int i=1; i < GRID_SIZE; i++){
-		if (maxNum > array[i][col]) {
-			maxNum = array[i][col];
+		if (maxNum > frame_conv[i][col]) {
+			maxNum = frame_conv[i][col];
 			maxRowId = i;
 		}
 	}
 	return maxRowId;
 }
 
+void convertToFrame(uint16_t COUNT_BUFF[], uint16_t ind) {
+	for (int i = 0; i < GRID_SIZE; i++) {
+		for (int j = 0; j < GRID_SIZE; j++) {
+			frame_conv[i][j] = COUNT_BUFF[(ind*NUM_PIXELS) + GRID_SIZE*i + j];
+		}
+	}
+	return;
+}
 
-bool determine_direction(uint16_t trigger_col, uint16_t offset, uint16_t *** COUNT_BUFF) {
-	if (TRIGGER_INDEX >= COUNT_BUFF_SIZE - 1 || TRIGGER_INDEX < 1){
+int determine_direction(uint16_t trigger_col, uint16_t offset, uint16_t COUNT_BUFF[]) {
+	if (TRIGGER_INDEX >= COUNT_BUFF_FRAMES - 1 || TRIGGER_INDEX < 1){
 		//exception
```

Failing silently is probably not a great idea for debugging.

---

Do you suggest using uart_out() for now here?

---

Yeah, that's fine.